From a0983d84185f04c4e40778fe951fde4439894882 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 16 Jul 2017 07:37:03 -0700
Subject: [PATCH] Remove assumptions about glibc being only libc
 implementation on linux

Signed-off-by: Khem Raj <raj.khem@gmail.com>
[Retrieved (and updated to work on current version) from:
http://cgit.openembedded.org/meta-openembedded/tree/meta-oe/recipes-support/open-vm-tools/open-vm-tools/0001-Remove-assumptions-about-glibc-being-only-libc-imple.patch?h=sumo]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 open-vm-tools/lib/file/fileIOPosix.c      | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

Index: open-vm-tools/lib/file/fileIOPosix.c
===================================================================
--- open-vm-tools.orig/lib/file/fileIOPosix.c
+++ open-vm-tools/lib/file/fileIOPosix.c
@@ -205,7 +205,7 @@ static AlignedPool alignedPool;
  * are not available in any header file.
  */
 
-#if defined(__linux__) && !defined(__ANDROID__)
+#if defined(__linux__) && defined(__GLIBC__)
    #if defined(_FILE_OFFSET_BITS) && (_FILE_OFFSET_BITS == 64)
       /*
        * We want preadv/pwritev. But due to FOB=64, the symbols are -64.
